### DISCUSSION TITLE
Support for Apollo Federation Gateway 2

### DISCUSSION THREAD

**Discussion by Vyse777** [Netflix/dgs-framework#988] [Ideas]
Is there any discussion going on yet for eventual support for Apollo Federation Gateway 2 recently released?
See here: https://www.apollographql.com/blog/announcement/backend/apollo-federation-2-is-now-generally-available/

DGS is listed as unsupported here: https://www.apollographql.com/docs/federation/other-servers/

Also, attempting this fails with an issue with DGS during runtime/startup: https://www.apollographql.com/docs/federation/federation-2/moving-to-federation-2/#add-schema-definitions-to-non-apollo-subgraphs

**Comment by srinivasankavitha** [comment]
We are aware of Apollo federation 2 and will be eventually switching to it once our internal gateway is ready as well (actively being explored). We won’t be able to switch our framework’s  federation libraries to support 2.0 just yet for the same reason, but will be very much  looking to support this. Timeline should be around later this year. We will also discuss if we can support this independent of our internal requirements. Will post updates on this issue as we have more concrete details.

**Reply by Vyse777** [reply]
Very awesome & exciting to hear! Thanks for the quick reply @srinivasankavitha

**Comment by babyfish-ct** [comment]
Agree, but not enough

Apollo's gateway requires hard code URL list of sub graphql services. 

It's not best solution for java/kotlin. Spring-Cloud use registry center, so the url list should be read from registry center such as eureka, dynamically.

**Reply by srinivasankavitha** [reply]
Internally, we do have our own implementation of a schema registry service and gateway that interact with each other and does what you describe.

**Reply by babyfish-ct** [reply]
Great! 

Take the liberty to ask, is it possible to make it public in the future?

**Reply by srinivasankavitha** [reply]
I believe the folks who own the gateway have considered it, but it's not a priority for the near term.

**Reply by setchy** [reply]
@babyfish-ct - Apollo Gateway when used with Managed Federation doesn't require a list of hardcoded subgraph endpoints...

**Reply by babyfish-ct** [reply]
Can it work with Spring Cloud's registry center(Not Apollo's registry center of Apollo's tech stack)?

**Reply by BCantos17** [reply]
Most likely not. For manage federation to work, apollo gateway requires specific environment variables that directs the gateway to Apollo Studios, you can try to make a feature request on apollo/gateway or apollo/router and see what they think
